include/quda_define.h.in

```diff
@@ -50,3 +50,17 @@
 #define FLOAT8
 #undef QUDA_FLOAT8
 #endif
+
+/**
+ * If we are building for a QUDA Device this will be set by CMAKE
+ */
+#cmakedefine QUDA_TARGET_CUDA
+#cmakedefine QUDA_TARGET_HIP
+
+#if defined(QUDA_TARGET_CUDA) && defined(QUDA_TARGET_HIP)
```

Comment:
Do we need this check at compile time? This file gets auto-generated and cmake won't define both values.

Reply:
Fair point (0f0198590)